Networking/NetworkingTests/ApplicationPassword/CookieNonceAuthenticatorTests.swift

```diff
@@ -0,0 +1,41 @@
+import XCTest
+@testable import Networking
+
+final class CookieNonceAuthenticatorTests: XCTestCase {
+
+    private let loginURL = URL(string: "https://example.com/wp-login.php")!
+    private let adminURL = URL(string: "https://example.com/wp-admin/")!
+    private let apiRequest = URLRequest(url: URL(string: "https://example.com/wp-json/")!)
+    private let sampleUser = "user123"
+    private let samplePassword = "password *+/$&=2+é"
+
+    func test_cookie_nonce_authenticator_encode_parameters_correctly() {
+        // Given
+        let config = CookieNonceAuthenticatorConfiguration(username: sampleUser,
+                                                           password: samplePassword,
+                                                           loginURL: loginURL,
+                                                           adminURL: adminURL)
+        let authenticator = CookieNonceAuthenticator(configuration: config)
+
+
+        let generatedBodyAsData = authenticator.authenticatedRequest().urlRequest!.httpBody!
+        let generatedBodyAsString = String(data: generatedBodyAsData, encoding: .utf8)
```

Comment:
Nit: we should use `try XCTUnwrap` instead of forced unwrapping to avoid crashing the test:

```suggestion
        let generatedBodyAsData = try XCTUnwrap(authenticator.authenticatedRequest().urlRequest?.httpBody)
        let generatedBodyAsString = try XCTUnwrap(String(data: generatedBodyAsData, encoding: .utf8))
```